doc/wg/core/notes/core-notes-2022-10-07.md

```diff
@@ -0,0 +1,391 @@
+   number
+ * Pat:
+   https://github.com/tock/tock/blob/master/doc/reference/trd104-syscalls.md#42-subscribe-class-id-1.
+   Reading very quickly, it looks like the TRD specifies what is done about
+   invalid upcall, but not invalid driver number?
+ * Pat: though maybe that's earlier..
+ * Johnathan: IDK what TRD 104 means, but that's what was stabilized in Tock 2.0
+ * Pat: If userspace invokes a system call on a peripheral driver that is not
+   installed in the kernel, the kernel MUST return a Failure result with an
+   error of NODEVICE. If userspace invokes an unrecognized system call on a
+   peripheral driver, the peripheral driver MUST return a Failure result with an
+   error of NOSUPPORT.
+ * Pat: That's in §4.0 of TRD104
+ * Pat: seems like we do have an implementation / documentation mismatch? :/
+ * Johnathan: Yes
+
+## Migrate `ProcessSlices` towards raw pointers #2977
+ * Leon: I've been looking at an ancient PR of mine, #2977, which tries to
+   migrate `ProcessSlices` towards raw pointers underneath. I think it's an
+   important issue to solve but tons of things have come up and I haven't had
+   time to get to it. `ProcessSlices` are currently built using `[Cell<>]` which
+   is unsound in terms of Rust's aliasing rules, and the only sound solution
+   seems to be to change the types to use raw pointers and use Rust's methods
+   for working on raw pointers. I think there is general agreement this is
+   useful and the API seems to be good. As part of the PR, we wanted to
+   transition as many capsules away from panicing on userspace buffer accesses
+   and perform more sensible error handling. I've spent a day rebasing it, and
+   reworked tons of capsules to refactor them. The lesson from that is we should
+   punt on converting capsules away from panicing accesses to a separate PR,
+   because it is hard to do that without changing their behavior on edge cases.
+   E.g. when a process asks a capsule to perform an operation that goes beyond
+   the bounds of the buffer it has shared. That's kind of the state of that, I
+   think the only thing missing is Miri tests, which I'm currently writing, to
+   have some confidence that what we're doing is sound. Does that make sense to
+   everybody?
+ * Pat: I think so
+ * Phil: It makes sense to me
+ * Alexandru: Yes it does
+ * Leon: The takeaway is I'm working on it again, I didn't forget it, it's just
+   a lot of work and as soon as we get the tests I propose to try to get it in.
+   Then work through capsules one-by-one to change them to sensible error
+   handling.
+ * Pat: It's a longstanding pain point so I'm onboard with getting this to
+   happen.
+
+## Component update PRs
+ * Pat: On the subject of large mechanical changes, Brad has a small army of
+   component PRs. I reviewed a bunch, tagged them last night. We'll probably
+   merge them now, unless someone has a reason not to, we should probably get
+   those in.
+ * Phil: I `bors r=`-ed the basic one.
```

Comment:
```suggestion
 * Phil: I `bors r+`-ed the basic one.
```

Reply:
Oops, somehow missed this for 2 weeks. Done!